File: spec/factories/habits.rb
# frozen_string_literal: true

FactoryBot.define do
  factory :habit do
    name { Faker::Hipster.sentence }
    repeat_interval { 1 }
    repeat_interval_unit { 'day' }
    start_date { Date.new(2021, 1, 6) }
    notify { false }
    user
  end
end

File: app/javascript/src/models/habits.ts
export type RepeatIntervalUnit = 'day' | 'week' | 'month' | 'business_day' | 'weekend' | 'weekday'

export interface Habit {
  id: number
  name: string
  repeatInterval: number
  repeatIntervalUnit: RepeatIntervalUnit
  startDate: string
  notify: boolean
  notificationTime: string | null
  deletedAt: string | null
  repeatSunday: boolean
  repeatMonday: boolean
  repeatTuesday: boolean
  repeatWednesday: boolean
  repeatThursday: boolean
  repeatFriday: boolean
  repeatSaturday: boolean
}

export const repeatIntervalUnits: RepeatIntervalUnit[] = ['business_day', 'weekend', 'weekday', 'day', 'week', 'month']
export const weekdayOptions: Array<keyof Habit> = ['repeatSunday', 'repeatMonday', 'repeatTuesday',
  'repeatWednesday', 'repeatThursday', 'repeatFriday', 'repeatSaturday']
